### ISSUE TITLE
"Add to cart" should be disabled after quantity of product added to cart is equal to available product quantity

### ISSUE THREAD

**Issue by asvishnyakov** [VirtoCommerce/vc-storefront-core#82]
Version info: 
- Browser version: Chrome 64
- Platform version: 2.13.21
- Storefront version: 3.0.10

### Expected behavior
Add to cart button should be disabled

### Actual behavior
Still enabled and successfully adds product to cart

### Steps to reproduce
1. Set product in stock quantity to 1 in Inventory blade on admin site
2. Go to storefront, add product to cart
3. Add to cart button still available even after page refresh, while on cart page you will get error "Product quantity exceeded"

**Comment by asvishnyakov** [comment]
Possible solution: [here](https://github.com/VirtoCommerce/vc-storefront-core/blob/2373dc580a7bcc14aa599ff6f02747125d540a3f/VirtoCommerce.Storefront/Domain/Catalog/CatalogService.cs#L109) should be a product quantity from cart instead of 1

**Comment by asvishnyakov** [comment]
@tatarincev ?

**Comment by tatarincev** [comment]
Yes right, the possible solution looks like this: just need to extend ctor of the `ProductIsAvailableSpecification` specification with the new  parameter `products in cart` and take into account it in  the specification

**Comment by asvishnyakov** [comment]
@tatarincev I think we just should pass product quantity in cart instead of 1

**Comment by asvishnyakov** [comment]
@tatarincev High priority

**Comment by Flanker72** [comment]
Related to a particular theme implementation. Current B2B functionality allows changing cart item quantity, so it will not be implemented.
